scene/2d/physics/collision_shape_2d.cpp

```diff
@@ -208,6 +208,24 @@ bool CollisionShape2D::is_disabled() const {
 	return disabled;
 }
 
+bool CollisionShape2D::contains_point(const Vector2 &p_point) const {
+	if (!collision_object)
+		return false;
+
+	Vector2 unscaledPoint;
+	Size2 scaleCheck;
+	if (collision_object->shape_owner_get_owner(owner_id)->is_class("Node2D")) {
+		Node2D *n = collision_object->shape_owner_get_owner(owner_id)->cast_to<Node2D>(collision_object->shape_owner_get_owner(owner_id));
+		unscaledPoint = (p_point - n->get_global_position()).rotated(n->get_global_rotation());
+		scaleCheck = n->get_global_scale();
+	} else {
+		Transform2D trans = collision_object->shape_owner_get_transform(owner_id);
+		unscaledPoint = (p_point - trans.get_origin()).rotated(trans.get_rotation());
+		scaleCheck = trans.get_scale();
+	}
+	return shape->contains_point(Vector2(unscaledPoint.x / scaleCheck.x, unscaledPoint.y / scaleCheck.y));
```

Comment:
These calculations look wrong and overcomplicated. But there is no need to do them manually, there are methods for transforming already available. I'd say it should be as simple as:
```suggestion
	return shape->contains_point(to_local(p_point));
```

Reply:
The calculations are correct, though they are very cumbersome you're right. Good to know there's a simpler alternative in the engine that can be used instead. Thanks for pointing it out!